gen/gen.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/fatih/structs"
+	mondoov1 "go.mondoo.com/mondoo-go"
+)
+
+func main() {
+	// Generate resources for Mondoo integrations only
+	err := generateIntegrationResources()
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+
+type IntegrationResource struct {
+	ResourceClassName     string
+	TerraformResourceName string
+}
+
+// generateIntegrationResources generates Terraform resources for Mondoo's integrations.
+func generateIntegrationResources() error {
+	// Read the template file
+	resourceTemplateFile := filepath.Join("gen", "templates", "integration_resource.go.tmpl")
+	resourceTmpl, err := template.ParseFiles(resourceTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	testTemplateFile := filepath.Join("gen", "templates", "integration_resource_test.go.tmpl")
+	testTmpl, err := template.ParseFiles(testTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	resourceDotTFTemplateFile := filepath.Join("gen", "templates", "resource.tf.tmpl")
+	resourceTFTmpl, err := template.ParseFiles(resourceDotTFTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	// Ensure the output directory exists
+	outputDirPath := filepath.Join("gen", "generated")
```

Comment:
Couldn't we create this directly in the directory the other integrations are in?

Reply:
not yet. still need some more work but we will ultimately write it in the right directory.

Reply:
We are creating it directly now!! 💯